nemo/nemo/core/neural_factory.py

```diff
@@ -319,6 +320,26 @@ def __init__(
                     backend="nccl", init_method="env://"
                 )
                 self._world_size = torch.distributed.get_world_size()
+
+                def torch_broadcast_wrapper(str_len=None, string=None, src=0):
```

Comment:
I realize that this is a larger issue, but I really don't love putting pytorch-specific code in neural_factory.py.  We are accumulating a lot of tech debt that we are going to have to clean up.

Reply:
yes, we have to replace stuff under "if backend == Backend.PyTorch:" with a function call kind of like we do it for PtActions (e.g. backend-specific). I'd say have an issue and do a separate PR for that.

Reply:
I saw that (import torch & co), but I still do not understand where we want to put the borderline between pieces that are agnostic from pieces that are framework-dependent...

Reply:
In general /nemo/nemo/core should not import torch or tf. Something to fix (in another PR), I believe this is the only place

Reply:
I think we all agree on this, resolving this conversation